Scripts/databasebackup.ps1

```diff
@@ -1,13 +1,49 @@
+#To be taken from pipeline need to change when pipeline values created 
+
+param (
+    [parameter(Mandatory=$true)]
+    [string]$lansaversion,
+
+    [parameter(Mandatory=$true)]
+    [string]$dbname,
+
+    [parameter(Mandatory=$true)]
+    [string]$location,
+
+    [parameter(Mandatory=$true)]
+    [string]$path,
+
+    [parameter(Mandatory=$true)]
+    [string]$databasestype,
+
+    [parameter(Mandatory=$true)]
+    [string]$servername
+)
 Set-AWSCredentials myAWScredentials
 
-$server = 'dbregressiontest.database.windows.net'
-$backupPath = 'C:\Program Files (x86)\AZURESQL'
-$s3bucket = 'lansa-us-east-1/db-regression-test/backups/150050/'
-$region = 'us-east-1'
+$server = $servername
+$backupPath = $path
+$s3bucket = 'lansa-us-east-1/db-regression-test/backups'
+$databasebackup = $s3bucket/$lansaversion/$databasestype
+$region = $location
+
+#$firewallrulename = $rulename
+#$currentrules = Get-AzSqlServerFirewallRule -ResourceGroupName $resourcegroup -ServerName $server -FirewallRuleName $firewallrulename
+#$serverip = $ip1, $ip2
 
 $databases = Invoke-Sqlcmd -ServerInstance $server -Username $user -Password $password -Query "SELECT [name]
-FROM master.dbo.sysdatabases where [name]='test'"
+FROM master.dbo.sysdatabases where [name]='$dbname'"
```

Comment:
username and password needs to be derived from AWS Secrets Manager.

Also this is Azure SQL Database specific, yet the script's name and parameters are written to be generic.

Reply:
will update the script to take the credentials from AWS secret manager

Reply:
Note that I am in the process of working out how to use a Service Principal to login. Up until now we have used a User and Password. With MFA forced on us that does not work in a batch script with no UI to validate the MFA.

I hope to get this addressed this evening.

Reply:
Hi Rob, 
we have figured how to connect to azure cloud using Service principal, we need a role assigned from Resource Group to service principal (sp). so, that the sp will be able to connect to the azure and subscription

Code snip:
`Connect-AzureRmAccount -ServicePrincipal -Credential $pscredential -Tenant 17e16064-c148-4c9b-9892-bb00e9589aa5`

Reply:
Thats correct. I had set one up on Saturday and forgot to hit send. Its in Teams, and here it is for you here too:

1. I have added azure-login.ps1 to the repo (you will need to rebase your changes in your PR on the latest state of the upline repo in order to get these changes). Use it with the params 'SP' '[password from AWSSecrets Manager]'
	where SP stands for Service Principal which is what I have created to enable hands free login now that MFA has had to be enabled on the account. Use this in the VM from the pipeline before executing scripts that need to use the Azure PowerShell CLI.
2. The Secret (UID & PWD) for the SP are in the Secret '[password/ServicePrincipalAzure](https://us-east-1.console.aws.amazon.com/secretsmanager/secret?name=password%2FServicePrincipalAzure&region=us-east-1)'
3. All the database UID/PWD are also in AWS Secrets Manager
4. This wiki describes the above https://github.com/lansa/db-regression/wiki/Passwords